services/management/ia/rules_service_test.go

```diff
@@ -0,0 +1,160 @@
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package ia
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/percona/pmm/api/managementpb"
+	iav1beta1 "github.com/percona/pmm/api/managementpb/ia"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"gopkg.in/reform.v1"
+	"gopkg.in/reform.v1/dialects/postgresql"
+
+	"github.com/percona/pmm-managed/models"
+	"github.com/percona/pmm-managed/utils/testdb"
+)
+
+func TestConvertTemplate(t *testing.T) {
+	ctx := context.Background()
+	sqlDB := testdb.Open(t, models.SkipFixtures, nil)
+	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+
+	// Enable IA
+	settings, err := models.GetSettings(db)
+	require.NoError(t, err)
+	settings.IntegratedAlerting.Enabled = true
+	err = models.SaveSettings(db, settings)
+	require.NoError(t, err)
+
+	testDir, err := ioutil.TempDir("", "")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err = os.RemoveAll(testDir)
+		require.NoError(t, err)
+	})
+
+	vmAlert := new(mockVmAlert)
+	vmAlert.On("RequestConfigurationUpdate").Return()
+
+	alertManager := new(mockAlertManager)
+	alertManager.On("RequestConfigurationUpdate").Return()
+
+	// Create channel
+	channels := NewChannelsService(db, alertManager)
+	respC, err := channels.AddChannel(context.Background(), &iav1beta1.AddChannelRequest{
+		Summary: "test channel",
+		EmailConfig: &iav1beta1.EmailConfig{
+			SendResolved: false,
+			To:           []string{"test@test.test"},
+		},
+		Disabled: false,
+	})
+	require.NoError(t, err)
+	channelID := respC.ChannelId
+	defer func() {
+		e := db.InTransaction(func(t *reform.TX) error {
+			return models.RemoveChannel(t.Querier, channelID)
+		})
+		assert.NoError(t, e)
+	}()
+
+	// Load test templates
+	templates := NewTemplatesService(db)
+	templates.userTemplatesPath = testUserTemplates
+	templates.Collect(ctx)
+
+	// Create test rule
+	rules := NewRulesService(db, templates, vmAlert, alertManager)
+	rules.rulesPath = testDir
+	resp, err := rules.CreateAlertRule(context.Background(), &iav1beta1.CreateAlertRuleRequest{
+		TemplateName: "user_rule",
+		Disabled:     false,
+		Summary:      "some testing rule",
+		Params: []*iav1beta1.RuleParam{
+			{
+				Name: "threshold",
+				Type: iav1beta1.ParamType_FLOAT,
+				Value: &iav1beta1.RuleParam_Float{
+					Float: 1.22,
+				},
+			},
+		},
+		For:      durationpb.New(2 * time.Second),
+		Severity: managementpb.Severity_SEVERITY_INFO,
+		CustomLabels: map[string]string{
+			"baz": "faz",
+		},
+		Filters: []*iav1beta1.Filter{
+			{
+				Type:  iav1beta1.FilterType_EQUAL,
+				Key:   "some_key",
+				Value: "60",
+			},
+		},
+		ChannelIds: []string{channelID},
+	})
+	require.NoError(t, err)
+	ruleID := resp.RuleId
+	defer func() {
+		e := db.InTransaction(func(t *reform.TX) error {
+			return models.RemoveRule(t.Querier, ruleID)
```

Comment:
`testdb.Open` recreates the whole database. There is no need to do manual cleanup.

Reply:
🤦‍♂️ removed